pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexCRUDViolationRule.java

```diff
@@ -337,6 +337,15 @@ private boolean isLastMethodName(final ASTMethodCallExpression methodNode, final
         return false;
     }
 
+    private boolean isWithSecurityEnforced(final AbstractApexNode<?> node){
+        if(node instanceof ASTSoqlExpression){
+            String pattern = "(?i).*[^']\\s*WITH SECURITY_ENFORCED\\s*[^']*";
+            String query = ((ASTSoqlExpression) node).getQuery();
+            return query.matches(pattern);
```

Comment:
Would `node.getNode().getQuery().endsWith("WITH SECURITY_ENFORCED");` produce the same result?

Reply:
I guess, we could also write `query.toUpperCase(Locale.ROOT).contains("WITH SECURITY_ENFORCED")`.
Btw. - are multiple spaces between "WITH" and "SECURITY_ENFORCED" allowed? If yes, then we probably need to use a regex. In that case, we should create a precompiled pattern constant.

Reply:
I see, we need to make sure, this text is not used inside a quoted string... So we'll need a regex.
Please create a private final static field:

```java
private static final Pattern WITH_SECURITY_ENFORCED = Pattern.compile("(?i).*[^']\\s*WITH\\s+SECURITY_ENFORCED\\s*[^']*");
```

And use it here then like this:

```java
return WITH_SECURITY_ENFORCED.matcher(soqlExpression.getQuery()).matches();
```